controllers/replica_set_controller.go

```diff
@@ -248,7 +248,8 @@ func (r ReplicaSetReconciler) Reconcile(ctx context.Context, request reconcile.R
 	return res, err
 }
 
-// updateLastSuccessfulConfiguration annotates the MongoDBCommunity resource with the latest configuration
+// updateLastSuccessfulConfiguration annotates the MongoDBCommunity resource with the configuration
+// that was used to create the resource
```

Comment:
> that was used to create the resource

This is called for every reconcile(whenever the spec is updated) -- why is it only annotating with the configuration that was used to create the resource?

Reply:
Good point